src/pending_transactions.erl

```diff
@@ -0,0 +1,124 @@
+%% @doc container handling pending transactions.
+%% 
+%% Transactions can come out of order.
+%% Newer transactions rewrite older ones.
+%% This structure allows purge of old transactions,
+%% to control the size of the buffer, however,
+%% it is not implemented here.
+
+-module(pending_transactions).
+
+-export([
+	 init/0,
+	 add_transaction/2,
+	 get_players/1,
+	 get_pending_transactions/2
+	]).
+
+-include_lib("stdlib/include/assert.hrl").
+
+-include("potato_records.hrl").
+
+%% @doc Initialize the container.
+
+init() ->
+    #pending_tx{player_map = maps:new(), counter = 0}.
+
+%% update Pending Transaction container with a new counter and a new transaction map for a player
+
+update_container(PlayerId, TxMp, NewCounter, PendingTx) -> 
+    %% erlang too hard
+
+    PM = (PendingTx#pending_tx.player_map)#{PlayerId => TxMp},
+    
+    PendingTx#pending_tx{player_map = PM, counter = NewCounter}.
+
+
+%% udpate a counter at a TxMap[Nonce] and return modified map and incremented counter
+
+update_counter_at_nonce(Nonce, {Counter, TxMap}) ->
+    {_OldCounter, Data} = maps:get(Nonce, TxMap),
+    TxMapNew = maps:put( {Counter, Data}, TxMap),
+    {Counter + 1, TxMapNew}.
+
```

Comment:
should be maps:put(Nonce, {Counter, Data}, TxMap)

Reply:
😱